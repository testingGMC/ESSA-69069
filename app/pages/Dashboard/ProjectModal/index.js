import { Form } from 'antd'
import ProjectModal from './ProjectModal'
export default Form.create({ name: 'manage_project' })(ProjectModal)
