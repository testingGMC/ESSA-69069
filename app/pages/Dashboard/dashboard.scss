.dashboard {

    .dashboard-content {

        padding: 2%;
        padding-left: 5%;
        padding-right: 5%;

        .no-members {
            display: flex;
            align-items: center;
            justify-content: center;
            height: 250px;
        }

        .project {
            padding-bottom: 2%;

            .project-title {
                font-weight: 700;
                font-size: 40px;
                color: black;
            }

            .right-side {
                display: flex;

                .button-add-project {
                    .ant-btn-primary {
                        color: #fff;
                        background-color: #2a4bd8 !important;
                        border-color: #1890ff;
                        text-shadow: 0 -1px 0 rgba(0, 0, 0, 0.12);
                        box-shadow: 0 2px 0 rgba(0, 0, 0, 0.045);
                    }

                    .plus-button {
                        margin-right: 20px;
                        margin-top: 10px;



                        .plus-icon {
                            font-size: 30px;
                            color: #2a4bd8;
                        }

                    }

                }

                .search-bar {
                    padding-bottom: 40px;
                    padding-top: 10px;
                }
            }

        }

        .row-style {
            padding-top: 30px;

        }

        .add-card-style {
            width: 90%;
            height: 140px;
            object-fit: contain;
            box-shadow: 0px 3px 8.1px 1.9px rgba(0, 0, 0, 0.07);
            border-color: blue;
            background-color: #ffffff;

            .project-plus-icon {
                font-size: 40px;
                color: #2a4bd8;
            }
        }
    }

}