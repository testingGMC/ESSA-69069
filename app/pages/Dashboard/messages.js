import { defineMessages } from 'react-intl'

export const scope = 'app.containers.Dashboard'

export default defineMessages({
  allproject: {
    id: `${scope}.allproject`,
    defaultMessage: 'All Projects',
  },
  search: {
    id: `${scope}.search`,
    defaultMessage: 'Search project name...',
  },
  createproject: {
    id: `${scope}.createProject`,
    defaultMessage: 'Create Project',
  },
  helmetDescription: {
    id: `${scope}.helmetDescription`,
    defaultMessage: 'Description of Dashboard',
  },
  helmetTitle: {
    id: `${scope}.helmetTitle`,
    defaultMessage: 'Dashboard',
  },
})
