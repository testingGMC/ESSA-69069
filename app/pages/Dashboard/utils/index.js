export const getFilteredProjects = (projects, searchValue) => {
    const value = searchValue.toLowerCase()
    return projects.filter(
        m =>
            m.name.toLowerCase().includes(value).toLowerCase().includes(value),
    )
}
