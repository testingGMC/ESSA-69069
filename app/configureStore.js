/**
 * Create the store with dynamic reducers
 */

import { createStore, applyMiddleware, compose } from 'redux'
import { routerMiddleware } from 'connected-react-router'
import createSagaMiddleware from 'redux-saga'
import { NOTIFICATION_TYPES } from 'shared/constants'
import { pushNotification } from 'shared/utils/notification'
import { errorsExtraction } from 'shared/utils/error-interceptors'
import reduxMiddlewares from './reduxMiddlewares'
import createReducer from './reducers'

export default function configureStore(initialState = {}, history) {
  let composeEnhancers = compose
  const reduxSagaMonitorOptions = {}

  // If Redux Dev Tools and Saga Dev Tools Extensions are installed, enable them
  /* istanbul ignore next */
  if (process.env.NODE_ENV !== 'production' && typeof window === 'object') {
    /* eslint-disable no-underscore-dangle */
    if (window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__)
      composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({})

    // NOTE: Uncomment the code below to restore support for Redux Saga
    // Dev Tools once it supports redux-saga version 1.x.x
    // if (window.__SAGA_MONITOR_EXTENSION__)
    //   reduxSagaMonitorOptions = {
    //     sagaMonitor: window.__SAGA_MONITOR_EXTENSION__,
    //   };
    /* eslint-enable */
  }

  // const error400Handling = store => next => action => {
  //   if (action.e && action.type.slice(-7) === 'FAILURE') {
  //     if (action.e.response.status === 400) {
  //       const errors = errorsExtraction(action.e.response.data.model)
  //       errors.forEach(error =>
  //         pushNotification(NOTIFICATION_TYPES.error, error),
  //       )
  //       next({ ...action, errors })
  //       return
  //     }
  //   }
  //   next(action)
  // }

  const sagaMiddleware = createSagaMiddleware(reduxSagaMonitorOptions)

  // Create the store with two middlewares
  // 1. sagaMiddleware: Makes redux-sagas work
  // 2. routerMiddleware: Syncs the location/URL path to the state
  const middlewares = [
    sagaMiddleware,
    routerMiddleware(history),
    ...reduxMiddlewares,
  ]

  const enhancers = [applyMiddleware(...middlewares)]

  const store = createStore(
    createReducer(),
    initialState,
    composeEnhancers(...enhancers),
  )

  // Extensions
  store.runSaga = sagaMiddleware.run
  store.injectedReducers = {} // Reducer registry
  store.injectedSagas = {} // Saga registry

  // Make reducers hot reloadable, see http://mxs.is/googmo
  /* istanbul ignore next */
  if (module.hot) {
    module.hot.accept('./reducers', () => {
      store.replaceReducer(createReducer(store.injectedReducers))
    })
  }

  return store
}
