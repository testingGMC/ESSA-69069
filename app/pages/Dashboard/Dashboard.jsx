/* eslint-disable global-require */
import React, { useState, useEffect } from 'react'
import PropTypes from 'prop-types'
import { Row, Col, Input, Button, Icon, Skeleton } from 'antd'
import { injectIntl, FormattedMessage, intlShape } from 'react-intl'
import { Helmet } from 'react-helmet'
import { isEmpty } from 'lodash'
import { isContentArchitect } from 'shared/utils/access-token'
import messages from './messages'
import './dashboard.scss'
import ProjectCard from './ProjectCard/ProjectCard'
import ProjectModal from './ProjectModal'

const { Search } = Input

const Dashboard = ({
  projects,
  fetchProjects,
  addProject,
  deleteProject,
  roles,
  intl,
  setIsVisible,
  updateProject,
  setIsUpdateModalVisible
}) => {
  const { isVisible } = projects
  const [cards, setCards] = useState(projects.data ? projects.data : [])
  const [formRef, setFormRef] = useState(null)
  const handleModalClose = () => {
    setIsVisible(false)
  }

  const handleSearch = value => {
    setCards(
      projects.data.filter(e =>
        e.name.toLowerCase().includes(value.toLowerCase()),
      ),
    )
  }

  const handleDelete = id => {
    deleteProject(id)
  }

  useEffect(() => {
    fetchProjects()
  }, [])

  useEffect(() => {
    if (projects.data) {
      setCards(projects.data)
    }
  }, [projects])

  const handleAddProject = () => {
    const { form } = formRef.props
    form.validateFields((err, values) => {
      if (err) {
        return
      }
      const newProject = {
        name: values.name,
        description: values.description,
        startDate: values.startDate,
        dueDate: values.dueDate,
      }
      addProject(newProject)
    })
  }

  const saveFormRef = formRefParameter => {
    setFormRef(formRefParameter)
  }
  const helmetMessages = {
    helmetTitle: intl.formatMessage({
      ...messages.helmetTitle,
    }),
    helmetDescription: intl.formatMessage({
      ...messages.helmetDescription,
    }),
  }

  return (
    <div className="dashboard">
      <Helmet>
        <title>{helmetMessages.helmetTitle}</title>
        <meta name="description" content={helmetMessages.helmetDescription} />
      </Helmet>

      <div className="dashboard-content">
        <Row>
          <Col>
            <Row gutter={48} align="bottom">
              <div className="project">
                <Col xs={24} sm={24} md={13} lg={14} xl={14} xxl={18}>

                  <div className="project-title">
                    <FormattedMessage {...messages.allproject} />
                  </div>
                </Col>
                <Col xs={24} sm={24} md={11} lg={10} xl={10} xxl={6}>
                  <div className="right-side">
                    <div className="button-add-project">
                      {!isEmpty(cards)
                        ? isContentArchitect(roles) && (
                          <div>
                            <Button
                              onClick={() => setIsVisible(true)}
                              className="plus-button"
                              type="primary"
                            >
                              Create Project
                            </Button>

                            {isVisible && (
                              <ProjectModal
                                wrappedComponentRef={saveFormRef}
                                isVisible={isVisible}
                                handleCancel={handleModalClose}
                                handleOk={handleAddProject}
                              />
                            )}
                          </div>
                        )
                        : ''}
                    </div>
                    <div className="search-bar">
                      <Search
                        placeholder={intl.formatMessage({
                          ...messages.search,
                        })}
                        onChange={e => handleSearch(e.target.value)}
                        style={{ width: 200 }}
                      />
                    </div>
                  </div>
                </Col>
              </div>
            </Row>
            <Row>
              <ProjectCard
                cards={cards}
                projects={projects}
                deleteProject={handleDelete}
                roles={roles}
                updateProject={updateProject}
                setIsUpdateModalVisible={setIsUpdateModalVisible}
              />
            </Row>
            {projects.loading ? (
              Array.from({ length: 6 }).map(() => (
                <Col xs={24} sm={24} md={8} lg={8} xl={8}>
                  <Skeleton loading={projects.loading} active avatar />
                </Col>
              ))
            ) : (
                !isEmpty(cards) ? ('') : (
                  isContentArchitect(roles) && (
                    <Row className="row-style">
                      <Col span={8} offset={8}>
                        <Button
                          type="dashed"
                          className="add-card-style"
                          size="large"
                          onClick={() => setIsVisible(true)}
                        >
                          <Icon
                            type="plus-circle"
                            theme="filled"
                            twoToneColor="#2a4bd8"
                            className="project-plus-icon"
                          />
                          <br />
                          <h3>
                            <FormattedMessage {...messages.createproject} />
                          </h3>
                        </Button>

                        {isVisible && (
                          <ProjectModal
                            wrappedComponentRef={saveFormRef}
                            isVisible={isVisible}
                            handleCancel={handleModalClose}
                            handleOk={handleAddProject}
                          />
                        )}
                      </Col>
                    </Row>
                  )
                )
              )}
          </Col>
        </Row>
      </div>
    </div>
  )
}

Dashboard.propTypes = {
  fetchProjects: PropTypes.func.isRequired,
  projects: PropTypes.object.isRequired,
  addProject: PropTypes.func.isRequired,
  deleteProject: PropTypes.func.isRequired,
  roles: PropTypes.array.isRequired,
  intl: intlShape.isRequired,
  setIsVisible: PropTypes.func.isRequired,
  updateProject: PropTypes.func.isRequired,
  setIsUpdateModalVisible: PropTypes.func.isRequired,
}

export default injectIntl(Dashboard)
