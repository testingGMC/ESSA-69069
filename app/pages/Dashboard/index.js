/**
 *
 * Dashboard
 *
 */

import React, { memo } from 'react'
import { connect } from 'react-redux'

import { createStructuredSelector } from 'reselect'
import { compose } from 'redux'
import injectSaga from 'utils/injectSaga'
import { DAEMON } from 'utils/constants'
import { useInjectReducer } from 'utils/injectReducer'
import makeSelectProject from '../../shared/redux/projects/selectors'
import reducer from '../../shared/redux/projects/reducer'

import * as actions from '../../shared/redux/projects/actions'

import saga from '../../shared/redux/projects/saga'

import Dashboard from './Dashboard'

const ProjectsIndex = props => {
  useInjectReducer({ key: 'projects', reducer })
  return <Dashboard {...props} />
}

const mapStateToProps = createStructuredSelector({
  projects: makeSelectProject(),
})

const mapDispatchToProps = {
  ...actions,
}

const withSaga = injectSaga({ key: 'projects', saga, mode: DAEMON })

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
)

export default compose(
  withConnect,
  memo,
  withSaga,
)(ProjectsIndex)
