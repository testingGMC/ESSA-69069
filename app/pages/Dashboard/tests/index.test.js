
import React from 'react'
import { render } from 'react-testing-library'
import { IntlProvider } from 'react-intl'
import { Provider } from 'react-redux'
import { browserHistory } from 'react-router-dom'

import ProjectsIndex from '../index'
import Dashboard from '../Dashboard'
import { DEFAULT_LOCALE } from '../../../i18n'
import configureStore from '../../../configureStore'


const initialProps = {
  fetchProjects: jest.fn(),
  addProject: jest.fn(),
  deleteProject: jest.fn(),
  projects: {}
}

describe('<ProjectsIndex />', () => {
  let store
  beforeAll(() => {
    store = configureStore({}, browserHistory)
  })

  it('Expect to not log in console', () => {
    const spy = jest.spyOn(global.console, 'log')
    render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <ProjectsIndex {...initialProps} />
        </IntlProvider>
      </Provider>
    )
    expect(spy).not.toHaveBeenCalled()
  })


  it('Expect to not log errors in console', () => {
    render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <ProjectsIndex {...initialProps} />
        </IntlProvider>
      </Provider>,
    )
    })

})

describe('<Dashboard/>', () => {
  let store
  beforeAll(() => {
    store = configureStore({}, browserHistory)
  })

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <Dashboard {...initialProps} />
        </IntlProvider>
      </Provider>,
    )
    expect(firstChild).toMatchSnapshot()

  })

  it('should fetch the Projects on mount if the projectId exists', () => {
    render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <Dashboard {...initialProps} projects />
        </IntlProvider>
      </Provider>,
    )
    expect(initialProps.fetchProjects).toHaveBeenCalled()
  })
})