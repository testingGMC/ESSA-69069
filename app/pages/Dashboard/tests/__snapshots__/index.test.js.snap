// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard/> should render and match the snapshot 1`] = `
<div
  class="dashboard"
>
  <div
    class="dashboard-content"
  >
    <div
      class="ant-row"
    >
      <div
        class="ant-col"
      >
        <div
          class="ant-row"
          style="margin-left: -24px; margin-right: -24px;"
        >
          <div
            class="project"
          >
            <div
              class="ant-col ant-col-xs-24 ant-col-sm-24 ant-col-md-15 ant-col-lg-17 ant-col-xl-18 ant-col-xxl-19"
              style="padding-left: 24px; padding-right: 24px;"
            >
              <div
                class="project-title"
              >
                <span>
                  All Projects
                </span>
              </div>
            </div>
            <div
              class="ant-col ant-col-xs-24 ant-col-sm-24 ant-col-md-9 ant-col-lg-7 ant-col-xl-6 ant-col-xxl-5"
              style="padding-left: 24px; padding-right: 24px;"
            >
              <div
                class="search-bar"
              >
                <span
                  class="ant-input-search ant-input-affix-wrapper"
                  style="width: 200px;"
                >
                  <input
                    class="ant-input"
                    placeholder="Search project name..."
                    type="text"
                    value=""
                  />
                  <span
                    class="ant-input-suffix"
                  >
                    <i
                      aria-label="icon: search"
                      class="anticon anticon-search ant-input-search-icon"
                      tabindex="-1"
                    >
                      <svg
                        aria-hidden="true"
                        class=""
                        data-icon="search"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                        />
                      </svg>
                    </i>
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-row"
        >
          <div>
            <div
              class="x-card"
            >
              <div
                class="no-members"
              >
                <div
                  class="ant-empty"
                >
                  <div
                    class="ant-empty-image"
                  >
                    <img
                      alt="No Data"
                      src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTg0IiBoZWlnaHQ9IjE1MiIgdmlld0JveD0iMCAwIDE4NCAxNTIiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+CiAgPGcgZmlsbD0ibm9uZSIgZmlsbC1ydWxlPSJldmVub2RkIj4KICAgIDxnIHRyYW5zZm9ybT0idHJhbnNsYXRlKDI0IDMxLjY3KSI+CiAgICAgIDxlbGxpcHNlIGZpbGwtb3BhY2l0eT0iLjgiIGZpbGw9IiNGNUY1RjciIGN4PSI2Ny43OTciIGN5PSIxMDYuODkiIHJ4PSI2Ny43OTciIHJ5PSIxMi42NjgiLz4KICAgICAgPHBhdGggZD0iTTEyMi4wMzQgNjkuNjc0TDk4LjEwOSA0MC4yMjljLTEuMTQ4LTEuMzg2LTIuODI2LTIuMjI1LTQuNTkzLTIuMjI1aC01MS40NGMtMS43NjYgMC0zLjQ0NC44MzktNC41OTIgMi4yMjVMMTMuNTYgNjkuNjc0djE1LjM4M2gxMDguNDc1VjY5LjY3NHoiIGZpbGw9IiNBRUI4QzIiLz4KICAgICAgPHBhdGggZD0iTTEwMS41MzcgODYuMjE0TDgwLjYzIDYxLjEwMmMtMS4wMDEtMS4yMDctMi41MDctMS44NjctNC4wNDgtMS44NjdIMzEuNzI0Yy0xLjU0IDAtMy4wNDcuNjYtNC4wNDggMS44NjdMNi43NjkgODYuMjE0djEzLjc5Mmg5NC43NjhWODYuMjE0eiIgZmlsbD0idXJsKCNsaW5lYXJHcmFkaWVudC0xKSIgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoMTMuNTYpIi8+CiAgICAgIDxwYXRoIGQ9Ik0zMy44MyAwaDY3LjkzM2E0IDQgMCAwIDEgNCA0djkzLjM0NGE0IDQgMCAwIDEtNCA0SDMzLjgzYTQgNCAwIDAgMS00LTRWNGE0IDQgMCAwIDEgNC00eiIgZmlsbD0iI0Y1RjVGNyIvPgogICAgICA8cGF0aCBkPSJNNDIuNjc4IDkuOTUzaDUwLjIzN2EyIDIgMCAwIDEgMiAyVjM2LjkxYTIgMiAwIDAgMS0yIDJINDIuNjc4YTIgMiAwIDAgMS0yLTJWMTEuOTUzYTIgMiAwIDAgMSAyLTJ6TTQyLjk0IDQ5Ljc2N2g0OS43MTNhMi4yNjIgMi4yNjIgMCAxIDEgMCA0LjUyNEg0Mi45NGEyLjI2MiAyLjI2MiAwIDAgMSAwLTQuNTI0ek00Mi45NCA2MS41M2g0OS43MTNhMi4yNjIgMi4yNjIgMCAxIDEgMCA0LjUyNUg0Mi45NGEyLjI2MiAyLjI2MiAwIDAgMSAwLTQuNTI1ek0xMjEuODEzIDEwNS4wMzJjLS43NzUgMy4wNzEtMy40OTcgNS4zNi02LjczNSA1LjM2SDIwLjUxNWMtMy4yMzggMC01Ljk2LTIuMjktNi43MzQtNS4zNmE3LjMwOSA3LjMwOSAwIDAgMS0uMjIyLTEuNzlWNjkuNjc1aDI2LjMxOGMyLjkwNyAwIDUuMjUgMi40NDggNS4yNSA1LjQydi4wNGMwIDIuOTcxIDIuMzcgNS4zNyA1LjI3NyA1LjM3aDM0Ljc4NWMyLjkwNyAwIDUuMjc3LTIuNDIxIDUuMjc3LTUuMzkzVjc1LjFjMC0yLjk3MiAyLjM0My01LjQyNiA1LjI1LTUuNDI2aDI2LjMxOHYzMy41NjljMCAuNjE3LS4wNzcgMS4yMTYtLjIyMSAxLjc4OXoiIGZpbGw9IiNEQ0UwRTYiLz4KICAgIDwvZz4KICAgIDxwYXRoIGQ9Ik0xNDkuMTIxIDMzLjI5MmwtNi44MyAyLjY1YTEgMSAwIDAgMS0xLjMxNy0xLjIzbDEuOTM3LTYuMjA3Yy0yLjU4OS0yLjk0NC00LjEwOS02LjUzNC00LjEwOS0xMC40MDhDMTM4LjgwMiA4LjEwMiAxNDguOTIgMCAxNjEuNDAyIDAgMTczLjg4MSAwIDE4NCA4LjEwMiAxODQgMTguMDk3YzAgOS45OTUtMTAuMTE4IDE4LjA5Ny0yMi41OTkgMTguMDk3LTQuNTI4IDAtOC43NDQtMS4wNjYtMTIuMjgtMi45MDJ6IiBmaWxsPSIjRENFMEU2Ii8+CiAgICA8ZyB0cmFuc2Zvcm09InRyYW5zbGF0ZSgxNDkuNjUgMTUuMzgzKSIgZmlsbD0iI0ZGRiI+CiAgICAgIDxlbGxpcHNlIGN4PSIyMC42NTQiIGN5PSIzLjE2NyIgcng9IjIuODQ5IiByeT0iMi44MTUiLz4KICAgICAgPHBhdGggZD0iTTUuNjk4IDUuNjNIMEwyLjg5OC43MDR6TTkuMjU5LjcwNGg0Ljk4NVY1LjYzSDkuMjU5eiIvPgogICAgPC9nPgogIDwvZz4KPC9zdmc+Cg=="
                    />
                  </div>
                  <p
                    class="ant-empty-description"
                  >
                    No Data
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
